### PR TITLE
fix(assets): remove componentWillReceiveProps

### PR DESCRIPTION
**Pull request by obdulia-losantos**
Due to: https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html

### CODE REVIEW

File: app/gather/assets/apps/components/PaginationBar.jsx [hunk 1]
```diff
@@ -84,9 +84,16 @@ class PaginationBar extends Component {
     }
   }
 
-  componentWillReceiveProps (nextProps) {
-    if (nextProps.currentPage !== this.state.currentPage) {
-      this.setState({ currentPage: Math.min(Math.max(1, nextProps.currentPage), this.getNumberOfPages()) })
+  getSnapshotBeforeUpdate (prevProps, prevState) {
```
Comment (admbtlr) [inline app/gather/assets/apps/components/PaginationBar.jsx]: Wouldn't it be better to use `getDerivedStateFromProps` here?

File: app/gather/assets/apps/components/PaginationContainer.jsx [hunk 2]
```diff
@@ -73,21 +73,28 @@ class PaginationContainer extends Component {
     return sortNumericArray([...new Set([...sizes, pageSize])]) // sort and remove duplicates
   }
 
-  componentWillReceiveProps (nextProps) {
-    this.setState({
-      sizes: this.buildSizes(nextProps.sizes, nextProps.pageSize)
-    }, () => {
-      if (nextProps.pageSize !== this.state.pageSize) {
-        this.setState({ pageSize: nextProps.pageSize, page: 1 })
+  getSnapshotBeforeUpdate (prevProps, prevState) {
```
Comment (admbtlr) [inline app/gather/assets/apps/components/PaginationContainer.jsx]: Same here...?

File: app/gather/assets/apps/survey/mask/SurveyMasks.jsx [hunk 3]
```diff
@@ -83,16 +83,20 @@ class SurveyMasks extends Component {
     }
   }
 
-  componentWillReceiveProps (nextProps) {
-    if (nextProps.survey !== this.props.survey ||
-      nextProps.columns !== this.props.columns) {
-      this.setState({
-        ...this.buildStateWithProps(nextProps, (nextProps.columns !== this.props.columns))
-      })
+  getSnapshotBeforeUpdate (prevProps) {
```
Comment (admbtlr) [inline app/gather/assets/apps/survey/mask/SurveyMasks.jsx]: And probably here too

**Review by admbtlr — CHANGES_REQUESTED** [review]

File: app/gather/assets/apps/components/PaginationBar.jsx (hunk 1 above)
Comment (obdulia-losantos) [reply app/gather/assets/apps/components/PaginationBar.jsx]: I tried but I could not make it work :disappointed_relieved: 

I can try it harder :hammer_and_wrench:

Comment (admbtlr) [reply app/gather/assets/apps/components/PaginationBar.jsx]: Weird.... it doesn't look _that_ complicated, although I've never used it before, so what do I know? 🤷‍♂

Comment (obdulia-losantos) [reply app/gather/assets/apps/components/PaginationBar.jsx]: OK, I finally manage to make it :sweat_smile:

**Review by admbtlr — APPROVED** [review]
😍

**Pull request merged**